### ISSUE TITLE
iOS: No OpKernel was registered to support Op 'Div' with these attrs

### ISSUE THREAD

**Issue by spdd** [tensorflow/tensorflow#3546]
I'm trying to build graph in iOS but i have following error:
```
Invalid argument: No OpKernel was registered to support Op 'Div' with these attrs
	 [[Node: conv1/conv/moments/sufficient_statistics/truediv = Div[T=DT_DOUBLE](conv1/conv/moments/sufficient_statistics/truediv/Cast, conv1/conv/moments/sufficient_statistics/truediv/Cast_1)]]
```

In Python if I run this code: 
`batch_mean, batch_var = tf.nn.moments(x, [0, 1, 2], keep_dims=True)`
I'm getting the error above, otherwise works fine.
I'm trying to find ops (core/ops, core/kernels).

**Comment by michaelisard** [comment]
This could be related to #3543 so I will update once we hear a response there.

**Comment by jbhav24** [comment]
I am having the same issue on iOS. Spent the entire day trying to fix it with no progress.

**Comment by spdd** [comment]
Any ideas?

**Comment by BarryTheBastard** [comment]
I'm experiencing same issue on Linux. Looks like in my case it only happens with int tensors.

**Comment by aselle** [comment]
You could try casting the quantities to float and see if that is supported? Otherwise, please provide the minimal and complete reproducible test case.. Here are some tests to try.

import tensorflow as tf.
a=tf.constant(3)
b=tf.constant(6)
sess=tf.Session()
sess.run(tf.truediv(a,b))
sess.run(tf.div(a,b))
sess.run(tf.truediv(af,bf))
sess.run(tf.div(af,bf))

**Comment by aselle** [comment]
Closing automatically due to lack of recent activity. Please reopen when further information becomes available. Thank you.

**Comment by cedricsimar** [comment]
For me, casting the input tensor to tf.float32 / tf.float64 before calling tf.nn.moments()  solved the issue

**Comment by StarRain-L** [comment]
@cedricsimar Your problem solved?I met a similar problem,and consult.Thank you!!

> No OpKernel was registered to support Op 'Maximum' with these attrs.  Registered devices: [CPU], Registered kernels:
  device='CPU'; T in [DT_FLOAT]

	 [[Node: gradients/Mean_grad/Maximum = Maximum[T=DT_INT32, _output_shapes=[[]], _device="/device:CPU:0"](gradients/Mean_grad/Prod_1, gradients/Mean_grad/Maximum/y)]]
